### PR TITLE
Job templates: Return full error message for invalid YAML

### PR DESCRIPTION
**Pull request by perlpunk**
YAML parsing errors are on multiple lines.

Issue: https://progress.opensuse.org/issues/66781

### CODE REVIEW

File: lib/OpenQA/YAML.pm [hunk 1]
```diff
@@ -91,8 +91,13 @@ sub validate_data {
...
     }
     catch {
-        # The first line of the backtrace gives us the error message we want
-        push @errors, (split /\n/, $_)[0];
+        if (m/^YAML::XS::Load Error/) {
+            push @errors, $_;
+        }
+        else {
+            # The first line of the backtrace gives us the error message we want
+            push @errors, (split /\n/, $_)[0];
```
Comment (Martchus) [inline lib/OpenQA/YAML.pm:99]: I suppose [using regex](https://github.com/os-autoinst/openQA/pull/3096/files#diff-2559347dfdfd46e3d678b4f8823f2d7fR577) would be faster.

Comment (perlpunk) [reply lib/OpenQA/YAML.pm:99]: Yeah, probably.
I didn't introduce this line, and it's only happening when something is wrong with the schema, so it should only happen in development anyway.

Comment (perlpunk) [reply lib/OpenQA/YAML.pm]: I added a LIMIT to the split. Ok?

**Comment by perlpunk** [comment]
I added a test for a non-existing schema file, as the code coverage was apparenly decreasing because I split up the `catch` block.

**Pull request merged**